document-portal/document-portal.c

```diff
@@ -1416,6 +1416,90 @@ portal_list (GDBusMethodInvocation *invocation,
   return TRUE;
 }
 
+const char *
+get_host_path_internal (GDBusMethodInvocation *invocation,
+             XdpAppInfo *app_info,
+             const char *id)
+{
+  g_autoptr(PermissionDbEntry) entry = NULL;
+
+  XDP_AUTOLOCK (db);
+
+  entry = permission_db_lookup (db, id);
+
+  if (!entry)
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             XDG_DESKTOP_PORTAL_ERROR, XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                                             g_strdup_printf("Invalid ID passed (%s)", id));
+      return NULL;
+    }
```

Comment:
maybe generate an error from the internal function and propagate it to dbus in  `portal_get_host_paths`?